### PR TITLE
Relax tightness of coupling in monitor code

### PR DESCRIPTION
**Pull request by prbans**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [ ] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.

### CODE REVIEW

**Comment by prbans** [comment]
image

File: azext_iot/tests/test_uamqp_import.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import mock
+
+
+class TestUamqpImport(object):
```
Comment (digimaun) [inline azext_iot/tests/test_uamqp_import.py:10]: Nice.

File: azext_iot/monitor/handlers/base_handler.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from abc import ABC, abstractmethod
+
+
+class AbstractBaseEventsHandler(ABC):
```
Comment (digimaun) [inline azext_iot/monitor/handlers/base_handler.py:10]: Good use of abstract base class.

**Comment by prbans** [comment]
image

File: azext_iot/monitor/handlers/_internal.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        return
+
+    parsed_msg = parser.parse_message(
+        msg,
+        pnp_context,
+        interface_name,
+        properties,
+        content_type,
+        simulate_errors,
+        central_device_provider,
```
Comment (digimaun) [inline azext_iot/monitor/handlers/_internal.py:40]: Ultimately the provider for device templates needs to be abstracted and not assume Central (there are services other than Central that deal with device templates).

File: azext_iot/monitor/handlers/handler.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        self,
+        device_id=None,
+        devices=None,
+        pnp_context=None,
+        interface_name=None,
+        content_type=None,
+        properties=None,
+        output=None,
+        validate_messages=None,
+        simulate_errors=None,
```
Comment (digimaun) [inline azext_iot/monitor/handlers/handler.py:26]: The named args list here is getting too long - we should move the less commonly used to kwargs.

File: azext_iot/monitor/handlers/_internal.py (hunk 3 above)
Comment (prbans) [reply azext_iot/monitor/handlers/_internal.py:40]: Agreed, I think that'll be the next item I'll address

File: azext_iot/monitor/handlers/handler.py (hunk 4 above)
Comment (prbans) [reply azext_iot/monitor/handlers/handler.py:26]: Agreed, I'll be addressing this as part of the next work item where I think I'll split the central parsing logic out so we dont have the _static_validation or _dynamic_validation calls in the common parser

**Pull request merged**